### ISSUE TITLE
Registry can't get BUCKET_NAME from etcd

### ISSUE THREAD

**Issue by Blystad** [deis/deis#3429]
This causes issues setting up registry on Amazon S3 instead of Ceph

Ref #2252 / #2812

**Comment by krancour** [comment]
@Blystad are you certain?

This looks to me like it does source the bucket name from etcd:

https://github.com/deis/deis/blob/master/registry/templates/config.yml

Am I missing something?

**Comment by Blystad** [comment]
It should, but it doesn't in the `boot` script. See: [https://github.com/deis/deis/blob/95746aadc8e2c77771174f2702dae6fc48951c4d/registry/bin/boot#L22,L54](https://github.com/deis/deis/blob/95746aadc8e2c77771174f2702dae6fc48951c4d/registry/bin/boot#L22,L54)

(Really just line 22, 45 and 54 which is interesting, but Github wouldn't let me mark those).

**Comment by Blystad** [comment]
This cause issues with `create_bucket` which attempts to create the S3 bucket if it doesn't exist.

**Comment by krancour** [comment]
Yup.  This is an issue alright.  I had looked at the boot script and really couldn't spot the problem in it.  _Now_ I see it... but only after banging my head against the wall for a while in #3431.  D'oh.  Sorry I doubted you, @Blystad.

I'd be happy to try and fix this issue, but I'd need a little guidance.  I'm not clear on why there's an environment variable involved here at all.  Any thoughts?  If I try to fix this, I want to get it right.

Ping @mboersma.  Any input?

**Comment by mboersma** [comment]
I'm not entirely sure why `BUCKET_NAME` is there for registry, if that's the env var you mean. It doesn't appear to be referenced elsewhere--maybe it was copy-pasted from similar code in database?

I think it would be safe to use the `BUCKET_NAME` env var if it's set, otherwise fetch the bucket name from etcd, otherwise default to "registry".

**Comment by krancour** [comment]
IMO, it's not worth even messing with a `BUCKET_NAME ` environment variable at all.  For someone to tune the value of an environment requires them to hack on a unit file and that just seems like a PIA compared to just setting the `bucketName` via deisctl.

How about this:

```
etcd_set_default bucketName registry
```

Then later in the boot script:

```
/app/bin/create_bucket $(etcdctl --no-sync -C ${ETCD} get ${ETCD_PATH}/registry)
```

Does that seem about right?  I did try this and I got a variety of errors, but I _think_ they are unrelated.  Trying it again now to be certain.

**Comment by krancour** [comment]
Ok... I end up with output like the following in the registry logs.  I can't quite tell if this error is related to the changes I made (as described above).  The bit about "WARNING Skipping confd config file" certainly sounds like a bad sign.  :(

```
-- Logs begin at Wed 2015-03-25 18:22:15 UTC. --
Apr 09 00:37:14 ip-10-0-3-235.us-west-1.compute.internal systemd[1]: Started deis-registry.
Apr 09 00:37:26 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: registry: waiting for confd to write initial templates...
Apr 09 00:37:31 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: Temporary debug info: Intended bucketName: <redacted, but is what I expect>
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: deis-registry running...
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 2015-04-09T00:37:32Z 4955c6b864db confd[55]: WARNING Skipping confd config file.
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:32 +0000] [60] [INFO] Starting gunicorn 19.1.1
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:32 +0000] [60] [INFO] Listening at: http://0.0.0.0:5000 (60)
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:32 +0000] [60] [INFO] Using worker: gevent
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:32 +0000] [67] [INFO] Booting worker with pid: 67
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:32 +0000] [68] [INFO] Booting worker with pid: 68
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:32 +0000] [69] [INFO] Booting worker with pid: 69
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:32 +0000] [70] [INFO] Booting worker with pid: 70
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:32 +0000 INFO: Enabling storage cache on Redis
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:32 +0000 INFO: Redis host: 10.0.3.235:6379 (db1)
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:32 +0000 INFO: Enabling lru cache on Redis
Apr 09 00:37:32 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:32 +0000 INFO: Redis lru host: 10.0.3.235:6379 (db2)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Enabling storage cache on Redis
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Redis config: {'path': '/', 'host': '10.0.3.235', 'password': None, 'db': 2, 'port': 6379}
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Configuring CORS for app:'docker-registry' with resources:[(<_sre.SRE_Pattern object at 0x7f6350501be8>, {})]
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Enabling storage cache on Redis
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Redis host: 10.0.3.235:6379 (db1)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Enabling lru cache on Redis
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Redis lru host: 10.0.3.235:6379 (db2)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Enabling storage cache on Redis
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Redis config: {'path': '/', 'host': '10.0.3.235', 'password': None, 'db': 2, 'port': 6379}
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Configuring CORS for app:'docker-registry' with resources:[(<_sre.SRE_Pattern object at 0x7f6350501be8>, {})]
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:33 +0000] [67] [ERROR] Exception in worker process:
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: Traceback (most recent call last):
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 507, in spawn_worker
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: worker.init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/ggevent.py", line 193, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(GeventWorker, self).init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 114, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.wsgi = self.app.wsgi()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/base.py", line 66, in wsgi
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.callable = self.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 65, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return self.load_wsgiapp()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return util.import_app(self.app_uri)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/util.py", line 356, in import_app
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: __import__(module)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/wsgi.py", line 20, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: from .tags import *  # noqa
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/tags.py", line 23, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: store = storage.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/storage/__init__.py", line 38, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: config=cfg)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/drivers/s3.py", line 72, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(Storage, self).__init__(path, config)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/core/boto.py", line 133, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self._boto_bucket = self._boto_conn.get_bucket(
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: AttributeError: 'NoneType' object has no attribute 'get_bucket'
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: Traceback (most recent call last):
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 507, in spawn_worker
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: worker.init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/ggevent.py", line 193, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(GeventWorker, self).init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 114, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.wsgi = self.app.wsgi()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/base.py", line 66, in wsgi
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.callable = self.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 65, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return self.load_wsgiapp()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return util.import_app(self.app_uri)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/util.py", line 356, in import_app
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: __import__(module)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/wsgi.py", line 20, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: from .tags import *  # noqa
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/tags.py", line 23, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: store = storage.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/storage/__init__.py", line 38, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: config=cfg)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/drivers/s3.py", line 72, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(Storage, self).__init__(path, config)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/core/boto.py", line 133, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self._boto_bucket = self._boto_conn.get_bucket(
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: AttributeError: 'NoneType' object has no attribute 'get_bucket'
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:33 +0000] [67] [INFO] Worker exiting (pid: 67)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:33 +0000] [68] [ERROR] Exception in worker process:
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: Traceback (most recent call last):
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 507, in spawn_worker
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: worker.init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/ggevent.py", line 193, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(GeventWorker, self).init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 114, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.wsgi = self.app.wsgi()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/base.py", line 66, in wsgi
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.callable = self.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 65, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return self.load_wsgiapp()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return util.import_app(self.app_uri)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/util.py", line 356, in import_app
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: __import__(module)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/wsgi.py", line 20, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: from .tags import *  # noqa
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/tags.py", line 23, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: store = storage.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/storage/__init__.py", line 38, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: config=cfg)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/drivers/s3.py", line 72, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(Storage, self).__init__(path, config)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/core/boto.py", line 133, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self._boto_bucket = self._boto_conn.get_bucket(
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: AttributeError: 'NoneType' object has no attribute 'get_bucket'
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: Traceback (most recent call last):
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 507, in spawn_worker
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: worker.init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/ggevent.py", line 193, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(GeventWorker, self).init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 114, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.wsgi = self.app.wsgi()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/base.py", line 66, in wsgi
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.callable = self.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 65, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return self.load_wsgiapp()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return util.import_app(self.app_uri)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/util.py", line 356, in import_app
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: __import__(module)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/wsgi.py", line 20, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: from .tags import *  # noqa
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/tags.py", line 23, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: store = storage.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/storage/__init__.py", line 38, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: config=cfg)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/drivers/s3.py", line 72, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(Storage, self).__init__(path, config)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/core/boto.py", line 133, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self._boto_bucket = self._boto_conn.get_bucket(
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: AttributeError: 'NoneType' object has no attribute 'get_bucket'
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:33 +0000] [68] [INFO] Worker exiting (pid: 68)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Enabling storage cache on Redis
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Redis host: 10.0.3.235:6379 (db1)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Enabling lru cache on Redis
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Redis lru host: 10.0.3.235:6379 (db2)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Enabling storage cache on Redis
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Redis config: {'path': '/', 'host': '10.0.3.235', 'password': None, 'db': 2, 'port': 6379}
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Configuring CORS for app:'docker-registry' with resources:[(<_sre.SRE_Pattern object at 0x7f6350501be8>, {})]
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Enabling storage cache on Redis
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Redis host: 10.0.3.235:6379 (db1)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Enabling lru cache on Redis
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Redis lru host: 10.0.3.235:6379 (db2)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Enabling storage cache on Redis
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Redis config: {'path': '/', 'host': '10.0.3.235', 'password': None, 'db': 2, 'port': 6379}
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: 09/Apr/2015:00:37:33 +0000 INFO: Configuring CORS for app:'docker-registry' with resources:[(<_sre.SRE_Pattern object at 0x7f6350501be8>, {})]
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:33 +0000] [69] [ERROR] Exception in worker process:
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: Traceback (most recent call last):
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 507, in spawn_worker
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: worker.init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/ggevent.py", line 193, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(GeventWorker, self).init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 114, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.wsgi = self.app.wsgi()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/base.py", line 66, in wsgi
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.callable = self.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 65, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return self.load_wsgiapp()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return util.import_app(self.app_uri)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/util.py", line 356, in import_app
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: __import__(module)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/wsgi.py", line 20, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: from .tags import *  # noqa
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/tags.py", line 23, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: store = storage.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/storage/__init__.py", line 38, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: config=cfg)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/drivers/s3.py", line 72, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(Storage, self).__init__(path, config)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/core/boto.py", line 133, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self._boto_bucket = self._boto_conn.get_bucket(
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: AttributeError: 'NoneType' object has no attribute 'get_bucket'
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: Traceback (most recent call last):
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 507, in spawn_worker
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: worker.init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/ggevent.py", line 193, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(GeventWorker, self).init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 114, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.wsgi = self.app.wsgi()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/base.py", line 66, in wsgi
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.callable = self.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 65, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return self.load_wsgiapp()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return util.import_app(self.app_uri)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/util.py", line 356, in import_app
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: __import__(module)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/wsgi.py", line 20, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: from .tags import *  # noqa
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/tags.py", line 23, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: store = storage.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/storage/__init__.py", line 38, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: config=cfg)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/drivers/s3.py", line 72, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(Storage, self).__init__(path, config)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/core/boto.py", line 133, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self._boto_bucket = self._boto_conn.get_bucket(
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: AttributeError: 'NoneType' object has no attribute 'get_bucket'
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:33 +0000] [69] [INFO] Worker exiting (pid: 69)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:33 +0000] [70] [ERROR] Exception in worker process:
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: Traceback (most recent call last):
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 507, in spawn_worker
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: worker.init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/ggevent.py", line 193, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(GeventWorker, self).init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 114, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.wsgi = self.app.wsgi()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/base.py", line 66, in wsgi
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.callable = self.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 65, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return self.load_wsgiapp()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return util.import_app(self.app_uri)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/util.py", line 356, in import_app
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: __import__(module)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/wsgi.py", line 20, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: from .tags import *  # noqa
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/tags.py", line 23, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: store = storage.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/storage/__init__.py", line 38, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: config=cfg)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/drivers/s3.py", line 72, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(Storage, self).__init__(path, config)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/core/boto.py", line 133, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self._boto_bucket = self._boto_conn.get_bucket(
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: AttributeError: 'NoneType' object has no attribute 'get_bucket'
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: Traceback (most recent call last):
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 507, in spawn_worker
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: worker.init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/ggevent.py", line 193, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(GeventWorker, self).init_process()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 114, in init_process
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.wsgi = self.app.wsgi()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/base.py", line 66, in wsgi
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self.callable = self.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 65, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return self.load_wsgiapp()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: return util.import_app(self.app_uri)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/gunicorn/util.py", line 356, in import_app
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: __import__(module)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/wsgi.py", line 20, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: from .tags import *  # noqa
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/tags.py", line 23, in <module>
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: store = storage.load()
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/storage/__init__.py", line 38, in load
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: config=cfg)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/drivers/s3.py", line 72, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: super(Storage, self).__init__(path, config)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: File "/usr/local/lib/python2.7/dist-packages/docker_registry/core/boto.py", line 133, in __init__
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: self._boto_bucket = self._boto_conn.get_bucket(
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: AttributeError: 'NoneType' object has no attribute 'get_bucket'
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:33 +0000] [70] [INFO] Worker exiting (pid: 70)
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:33 +0000] [60] [INFO] Shutting down: Master
Apr 09 00:37:33 ip-10-0-3-235.us-west-1.compute.internal sh[1228]: [2015-04-09 00:37:33 +0000] [60] [INFO] Reason: Worker failed to boot.
Apr 09 00:37:45 ip-10-0-3-235.us-west-1.compute.internal systemd[1]: deis-registry@1.service: main process exited, code=exited, status=1/FAILURE
Apr 09 00:37:45 ip-10-0-3-235.us-west-1.compute.internal docker[1641]: Error response from daemon: No such container: deis-registry
Apr 09 00:37:45 ip-10-0-3-235.us-west-1.compute.internal docker[1641]: time="2015-04-09T00:37:45Z" level="fatal" msg="Error: failed to remove one or more containers"
Apr 09 00:37:45 ip-10-0-3-235.us-west-1.compute.internal systemd[1]: Unit deis-registry@1.service entered failed state.
Apr 09 00:37:45 ip-10-0-3-235.us-west-1.compute.internal systemd[1]: deis-registry@1.service failed.
Apr 09 00:37:51 ip-10-0-3-235.us-west-1.compute.internal systemd[1]: deis-registry@1.service holdoff time over, scheduling restart.
Apr 09 00:37:51 ip-10-0-3-235.us-west-1.compute.internal systemd[1]: Stopping deis-registry...
```

**Comment by krancour** [comment]
"AttributeError: 'NoneType' object has no attribute 'get_bucket'"

^ Looks like maybe some other bit of code further on down the line might not be learning about the bucket name now.

**Comment by Blystad** [comment]
I've fought with this once already. If it cannot connect to s3 it returns a NoneType (null)...

Did you set s3region in etcd? That was my error

**Comment by krancour** [comment]
@Blystad good call.  That was it.  I had actually seen advice in another issue to _not_ set that, although that was a very old, pre-1.0 issue.  I get much farther when I do this.

At this point, I'm confident that my proposed fix works.  I'll get to work on a PR.

That being said, I still hit an S3 issue.  It seems the registry (tries) to sync to my bucket in S3 at startup, but it doesn't get very far because it keeps getting 404s.  @Blystad any chance you ran into that one before as well?

**Comment by krancour** [comment]
Looks like data didn't copy up to my bucket properly in the first place.  That probably explains it.  Will move ahead with the PR.

**Comment by krancour** [comment]
FYI... #3445 was the cause of latest issue.

**Comment by Blystad** [comment]
I had actually found the same docs that you found krancour. But, (and one could argue that this is a bug), in the `config.yml` file, if s3Region is not set then it sets it to `""` but in the YAML standard a null value is `~`. Since the python code checks "if s3Region is not null, then connect using the provided region", that fails since the s3Region isn't null (it's `""`), but it isn't a valid s3Region either (like `eu-west-1`).

I found that by starting the `deis-registry` container and walking through the boot script and various configuration scripts.
